Lib/test/test_calendar.py

```diff
@@ -493,6 +493,7 @@ def test_format(self):
 class CalendarTestCase(unittest.TestCase):
 
     def test_deprecation_warning(self):
+        warnings.resetwarnings()
```

Comment:
Why aren't we using `self.assertWarns`?

Reply:
I don't have a reason not to use `self.assertWarns`. I just made the minimal change to fix the issue. Looking through the previous PR, it doesn't look like using `assertWarns(Regex)` was suggested, so the contributor may not have been aware of it.

Reply:
Let's switch to `assertWarnsRegex`, then we don't have to do all this manual manipulation of the warnings machinery.

Reply:
yeah, sounds good